macros/contextFraction.pl

```diff
@@ -905,7 +905,7 @@ sub string {
 sub TeX {
   my $self = shift; my $equation = shift; my $prec = shift;
   my ($a,$b) = @{$self->{data}}; my $n = "";
-  return $a if $b == 1;
+  return "$a" if $b == 1;
   if ($self->getFlagWithAlias("showMixedNumbers","showProperFractions") && CORE::abs($a) > $b)
     {$n = int($a/$b); $a = CORE::abs($a) % $b; $n .= " " unless $a == 0}
```

Comment:
This assignment to `$n` can also lead to a numeric result rather than a string (when `$a` is a non-zero multiple of `$b` and parentheses are not needed around the fraction).

There are similar issues with the `string()` method above (both the original problem with `$a` and this one with `$n`) that should also be patched.

Reply:
excellent - I have pushed an update to ensure that `$a` and `$n` are returned as strings in both `TeX()` and `string()`